### PR TITLE
:technologist: add dev container

### PR DESCRIPTION
**Pull request by m4nch0t**

### CODE REVIEW

File: .devcontainer/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+    sudo tar -xzf /tmp/crane.tar.gz -C /usr/local/bin/ crane && \
+    rm -rf /tmp/crane.tar.gz
+
+# Install go
+ARG GO_VERSION=1.20.14
+
+RUN curl --fail -L -o /tmp/go.tar.gz https://go.dev/dl/go${GO_VERSION}.linux-amd64.tar.gz && \
+    sudo tar -zxvf /tmp/go.tar.gz -C /usr/local/bin/ go/bin/go --strip-components 2 && \
+    rm -rf /tmp/go.tar.gz
```
Comment (eg-ayoub) [inline .devcontainer/Dockerfile]: maybe use 1.19 like for operator 
codegen needs to be tested with this

Comment (m4nch0t) [reply .devcontainer/Dockerfile]: as discussed : 
- 1.19 -->127
- 1.20 --> 128

File: .devcontainer/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+ARG GO_VERSION=1.20.14
+
+RUN curl --fail -L -o /tmp/go.tar.gz https://go.dev/dl/go${GO_VERSION}.linux-amd64.tar.gz && \
+    sudo tar -C /usr/local -xzf /tmp/go.tar.gz && \
+    sudo ln -s /usr/local/go/bin/go /usr/local/bin/go && \
+    sudo ln -s /usr/local/go/bin/gofmt /usr/local/bin/gofmt && \
+    rm -rf /tmp/go.tar.gz
+
+# Install golangci-lint
+ARG GOLANGCI_VERSION=1.53.3
```
Comment (TeddyAndrieux) [inline .devcontainer/Dockerfile]: nit: 1.54.2 to be consistent with what we use in the operator

File: .github/workflows/generate-sbom.yaml [hunk 3]
```diff
@@ -9,7 +9,7 @@ on:
         type: string
       artifacts-url:
         description: >-
-          URL from which to retrieve an OS image and all ARTESCA artifacts
+          URL from which to retrieve an OS image and all metalk8s artifacts
```
Comment (TeddyAndrieux) [inline .github/workflows/generate-sbom.yaml:12]: :roll_eyes:

**Review by TeddyAndrieux — DISMISSED** [review]

File: .devcontainer/Dockerfile (hunk 2 above)
Comment (m4nch0t) [reply .devcontainer/Dockerfile]: All right!

**Review by TeddyAndrieux — APPROVED** [review]

**Review by eg-ayoub — APPROVED** [review]

**Pull request merged**